src/data/list/rotate.lean

```diff
@@ -96,6 +96,10 @@ by rw [rotate_eq_rotate', rotate'_length]
 @[simp] lemma rotate_length_mul (l : list α) (n : ℕ) : l.rotate (l.length * n) = l :=
 by rw [rotate_eq_rotate', rotate'_length_mul]
 
+/-- A version of `rotate_length_mul` useful for rewriting. -/
+lemma rotate_length_mul' (l : list α) (m n : ℕ) (h : l.length = m) : l.rotate (m * n) = l :=
+h ▸ rotate_length_mul _ _
+
```

Comment:
This feels like a weird lemma to me.

Reply:
I use it further down in `zip_with_rotate_distrib` in a rewrite. Getting the multiplicands to be in the precise form of the list is hard, when the list is some chain of list operations.

Reply:
I think this might be a nicer lemma to solve that problem:
```
lemma rotate_eq_take_append_drop' {l : list α} {n : ℕ} :
  l.rotate n = l.drop (n % l.length) ++ l.take (n % l.length) :=
begin
  cases h : l.length with m,
  { rw list.length_eq_zero.mp h,
    simp, },
  have : n = n % l.length + l.length * (n / l.length) := (nat.mod_add_div _ _).symm,
  conv_lhs { rw this },
  rw [add_comm, ←rotate_rotate, rotate_length_mul, ←h],
  refine rotate_eq_take_append_drop (n.mod_lt $ _).le,
  exact h.symm ▸ m.succ_pos,
end
```